rspec-tools/rspec_tools/validation/description.py

```diff
@@ -98,3 +186,28 @@ def validate_source_language(rule_language: LanguageSpecificRule):
         elif not known_highlight(pre.code['data-lang']):
           raise RuleValidationError(f'''Rule {rule_language.id} has unknown language "{pre.code['data-lang']}" in code example in section "{name}".
 Are you looking for "{highlight_name(rule_language)}"?''')
+
+def validate_subsections(rule_language: LanguageSpecificRule):
+  for optional_section in list(OPTIONAL_SECTIONS.keys()):
+    validate_subsections_for_map(rule_language, optional_section, OPTIONAL_SECTIONS)
+  for mandatory_section in list(SECTIONS.keys()):
+    if mandatory_section == HOW_TO_FIX_IT:
+      validate_subsections_for_map(rule_language, mandatory_section, SECTIONS, HOW_TO_FIX_IT_REGEX)
+    else:
+      validate_subsections_for_map(rule_language, mandatory_section, SECTIONS)
+
+def validate_subsections_for_map(rule_language: LanguageSpecificRule, section_name: str, map, section_regex = None):
```

Comment:
```suggestion
def validate_subsections_for_map(rule_language: LanguageSpecificRule, section_name: str, allowed_subsections: set[str], section_regex = None):
```
this makes it more explicit, and reduces the number of lookups to the `SECTIONS` map

Reply:
done in e45c99c72